src/controllers/roomController.js

```diff
@@ -0,0 +1,110 @@
+    }
+  }
+
+  /**
+   * Get all  rooms
+   * @param {Object} req provides the requests from Admin to controllers
+   * @param {Object} res provides responses to the admin
+   * @return {object} Oject of data or error
+  */
+  static async getAllRooms(req, res) {
+    try {
+      const availableRooms = await RoomServices.findAllRooms();
+      if (availableRooms.length === 0) return Response.error(res, code.notFound, 'No Rooms found in the accommodations');
+      return Response.success(res, code.ok, 'Successfuly fetched all rooms', { rooms: availableRooms });
+    } catch (err) {
+      return Response.error(res, code.serverError, 'Something went wrong');
+    }
+  }
+
+  /**
+   * Get single  room
+   * @param {Object} req provides the requests from Admin to controllers
+   * @param {Object} res provides responses to the admin
+   * @return {object} Oject of data or error
+  */
+  static async getSingleRoom(req, res) {
+    try {
+      const { id } = req.params;
+      const room = await RoomServices.findSingleRooms({ id });
+      if (!room) return Response.error(res, code.notFound, 'Room does not exist');
+      return Response.success(res, code.ok, 'Successfuly fetched single room', { room });
+    } catch (err) {
+      return Response.error(res, code.serverError, 'Something went wrong');
+    }
+  }
+
+  /**
+   * delete single  room
+   * @param {Object} req provides the requests from Admin to controllers
+   * @param {Object} res provides responses to the admin
+   * @return {object} Oject of data or error
+  */
+  static async deleteRoom(req, res) {
+    try {
+      const { id } = req.params;
+      const room = await RoomServices.findSingleRooms({ id });
+      if (!room) return Response.error(res, code.notFound, 'Room does not exist');
+      const deletedRoom = await RoomServices.deleteRoom({ id });
+      if (!deletedRoom) return Response.error(res, code.notFound, 'Failed to delete a room');
+      return Response.success(res, code.ok, 'Successfuly deleted room');
+    } catch (err) {
+      return Response.error(res, code.serverError, 'Something went wrong');
+    }
+  }
+
+  /**
+   * update room
+   * @param {Object} req provides the requests from Admin to controllers
+   * @param {Object} res provides responses to the admin
+   * @return {object} Oject of data or error
+  */
+  static async updateRoom(req, res) {
+    try {
+      const { id } = req.params;
+      const room = await RoomServices.findSingleRooms({ id });
+      if (!room) return Response.error(res, code.notFound, 'Room does not exist');
+      if (req.accommodation.id !== room.accommodationId) return Response.error(res, code.notFound, 'Room does not exist in this accommodation');
+      // uploading pictures
+      const roomImages = await uploadPic(req.files, res);
+      if (roomImages.length === 0) return Response.error(res, code.notFound, 'please! provide a picture for your room');
```

Comment:
is it necessary to update a room picture? @muchristian Kindly in while updating image is optional.